hack/make-rules/test.sh

```diff
@@ -120,10 +120,6 @@ KUBE_TEST_API_VERSIONS="${KUBE_TEST_API_VERSIONS:-${ALL_VERSIONS_CSV}}"
 # once we have multiple group supports
 # Create a junit-style XML test report in this directory if set.
 KUBE_JUNIT_REPORT_DIR=${KUBE_JUNIT_REPORT_DIR:-}
-# If KUBE_JUNIT_REPORT_DIR is unset, and ARTIFACTS is set, then have them match.
```

Comment:
I don't understand why this is being removed.

Reply:
Took me some time, but I think I finally understood what's the issue.
The problem is cause by https://github.com/kubernetes/kubernetes/blob/a3c14ecfa5f463bf712760e7b823ecd32cefca4b/hack/make-rules/test.sh#L185 and to be more precise https://github.com/kubernetes/kubernetes/blob/a3c14ecfa5f463bf712760e7b823ecd32cefca4b/hack/make-rules/test.sh#L188

When junit reporter is enabled, it filters out most of the logs.
I see two possible fixes:
- remove setting KUBE_JUNIT_REPORT_DIR to ARTIFACTS as default
- remove this go_test_grep_pattern that filters out logs

Reply:
Hmm, we probably don't want to remove this, as this is an important system in our CI system.

Reply:
If ARTIFACTS is also unset, then this won't be set (the local use case).

This is important for CI and shouldn't be removed.
/lgtm cancel

Reply:
Ok, so lets proceed then with changing `go_test_grep_pattern`.
I added flag that will force printing full log. `hack/jenkins/benchmark-dockerized.sh` will set this flag, get full log and convert it into benchmark. At the same time, there should be no impact on your CI.
Wdyt?